core/src/main/java/org/apache/mesos/offer/DefaultOfferRequirementProvider.java

```diff
@@ -155,4 +156,35 @@ private void validateVolumes(Protos.TaskInfo taskInfo, TaskSpecification taskSpe
 
         return volumes;
     }
+
+    /**
+     * Creates an {@link Protos.org.apache.mesos.Protos.ExecutorInfo} that drives
+     * the {@link org.apache.mesos.executor.CustomExecutor}.
+     * @param taskSpecification The {@link TaskSpecification} used to setup relevant environment for the executor.
+     * @return The {@link org.apache.mesos.Protos.ExecutorInfo} to run
+     * the {@link org.apache.mesos.executor.CustomExecutor}
+     * @throws NullPointerException
+     */
+    private Protos.ExecutorInfo getExecutorInfo(TaskSpecification taskSpecification) throws NullPointerException {
+
+        Protos.CommandInfo.URI executorURI;
+        Protos.ExecutorInfo.Builder executorInfoBuilder = Protos.ExecutorInfo.newBuilder()
+                .setName(taskSpecification.getName())
+                .setExecutorId(Protos.ExecutorID.newBuilder().setValue("").build()); // Set later by ExecutorRequirement
+
+        try {
+            executorURI = TaskUtils.uri(System.getenv(EXECUTOR_URI));
+        } catch (NullPointerException e) {
+            throw new NullPointerException(String.format("Failed at setting the URI for the executor: %s", e));
```

Comment:
This feels like taking advantage of a side effect, how about a more explicit check like...:
```
String executorStr = System.getenv(EXECUTOR_URI);
if (executorStr == null) {
  throw new IllegalStateException("Missing environment variable: " + EXECUTOR_URI);
}
executorUri = TaskUtils.uri(executorStr);
```